ASFAchievementManager/AchievementHandler.cs

```diff
@@ -90,16 +96,18 @@ public sealed class AchievementHandler : ClientMsgHandler {
 							if (int.TryParse(Achievement.Name, out int bitNum)) {
 								if (uint.TryParse(stat.Name, out uint statNum)) {
 									bool isSet = false;
-									if ((Response.stats != null) && (Response.stats.Find(statElement => statElement.stat_id == statNum) != null)) {
-										isSet = (Response.stats.Find(statElement => statElement.stat_id == statNum)!.stat_value & ((uint) 1 << bitNum)) != 0;
+									uint stat_value = 0;
+									if (Response?.stats?.Find(statElement => statElement.stat_id == statNum) != null) {
+										stat_value = Response.stats.Find(statElement => statElement.stat_id == statNum)!.stat_value;
```

Comment:
Evaluating this `Find` twice is a waste of time.
something like `uint? stat_value = Response?.stats?.Find(statElement => statElement.stat_id == statNum)?.stat_value;` and then `bool isSet = stat_value != null && (stat_value! & ((uint) 1 << bitNum)) != 0` should do the trick. Don't forget to use the `!` operator further down when accessing the value.

Reply:
Oh past lazy me, forgot I still had that like that. Updated!